### DISCUSSION TITLE
Upgrading Old Site

### DISCUSSION THREAD

**Discussion by kegs85** [pradosoft/prado#990] [Q&A]
I'm midway through an upgrade of a very old site (Prado 3.1.2) to 4.2.2 but have a dependency on some older js packages,
I dont want to have to update all the js components right now and would like to reinstate prototype (1.6.0.2) support but only seem to get this working in debug mode - is there something I'm missing?

**Comment by ctrlaltca** [comment]
I would suggest you to make the update one step at a time, targeting prado 3.2 first and following these directions: http://www.pradoframework.net/demos/quickstart/?page=GettingStarted.Upgrading32
Once you get that working, you can target prado 4.x.

Also, remember to always clean your asset/ and protected/runtime folders to ensure you don't get server old  version of the js files.
What error do you get exactly?

**Reply by kegs85** [reply]
The behavior I am seeing is that when debug mode is inactive, it appears the prototype.js is being ignored, and if it is active, it is recognising the functions.  I get the same errors outside debug mode as I do when I remove the script tag to include prototype.js.

I might have to get our functions updated to avoid prototype, just trying to minimise what we need to do right now

**Comment by kegs85** [comment]
Thanks, whilst I have upgraded the framework as one jump, I have gone through all required steps of the progressive upgrade instructions to target affected code, as well as having update our php base version to 8.2 - generally speaking things are working, but the main issue here is affecting the drop down menus, and the different behavior of javascript can be best seen below.  

In normal mode:
![image](https://github.com/pradosoft/prado/assets/86577818/24d3cb36-0f5d-4630-a350-379390359a19)

In debug mode:
![image](https://github.com/pradosoft/prado/assets/86577818/db7f75e3-d715-4a30-92e4-df2eddc5f568)

**Comment by ctrlaltca** [comment]
Can you check in the "sources" tab of the developer tools what javascript files gets actually loaded? You should find them inside assets/*

**Reply by kegs85** [reply]
Apologies for not returning/responding earlier.
We ended up rewriting/updating our custom JS to be jQuery 3.x compliant, and ran in no conflict mode whilst progressing through the upgrade.
